### PR TITLE
Windows advanced debugging documentation.

### PR DESCRIPTION
**Pull request by CIPop**
Adding documentation on how to use WinDBG as debugger for CoreFX tests and applications.

@davidsh @ericstj @weshaggard @stephentoub PTAL

### CODE REVIEW

File: Documentation/debugging/windows-instructions.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+Assuming that your repo is at `C:\corefx`:
+
+```
+cd C:\corefx\bin\tests\Windows_NT.AnyCPU.Debug\System.Net.Sockets.Tests\dnxcore50
+C:\corefx\bin\tests\Windows_NT.AnyCPU.Debug\System.Net.Sockets.Tests\dnxcore50\CoreRun.exe xunit.console.netcore.exe System.Net.Sockets.Tests.dll -xml testResults.xml -notrait category=nonwindowstests -notrait category=OuterLoop -notrait category=failing
+```
+
+* If the test crashes or encounteres a `Debugger.Launch()` method call, WinDBG will automatically start and attach to the `CoreRun.exe` process 
+
```
Comment (davidsh) [inline Documentation/debugging/windows-instructions.md]: Do we need to do anything special to move the debugger from the parent process (CoreRun.exe) to the child process process (xunit.console.netcore.exe)?  Or is there even a child process?  Perhaps xunit is treated as a .NET assembly that is just loaded and then some starting class/method is launched in the app domain..

Comment (CIPop) [inline Documentation/debugging/windows-instructions.md]: This is an interesting question. The short answer is that nothing special is required.

Looking at my debugging session, as far as the OS is concerned, there is only one process: CoreRun.exe. Xunit assemblies have been loaded by the runtime but none of them are separate processes or AppDomains. The process has a single AppDomain created by CoreRun.

This is seems to be very familiar to how the old CLR was executing.

Comment (mellinoe) [inline Documentation/debugging/windows-instructions.md]: There is no child process, xunit.console.netcore.exe is just a regular .NET executable assembly that CoreCLR executes, and the runner executes all of the tests in-process.

File: Documentation/debugging/windows-instructions.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
+Debugging CoreFX on Windows
+==========================
+
+You can Debug .NET Core via Visual Studio or WinDBG.
+
+For Visual Studio debugging, follow the instructions at [Debugging tests in Visual Studio](https://github.com/dotnet/corefx/blob/master/Documentation/building/windows-instructions.md) to run and debug tests.
```
Comment (stephentoub) [inline Documentation/debugging/windows-instructions.md]: On the referenced windows-instructions page, can you add a link back to this page as well?

File: Documentation/debugging/windows-instructions.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+1. Install [PerfView](http://www.microsoft.com/en-us/download/details.aspx?id=28567)
+2. Run PerfView as Administrator
+3. Disable all other collection parameters
+4. Add Additional Providers (see below - Important: keep the "*" wildcard before the names.)
+
+### Tracing within System.Net namespaces
+
+* `*Microsoft-System-Net-Debug`: Highly verbose, low-level debugging traces in production code.
+* `*Microsoft-System-Net-TestLogging`: Test-code tracing (I/O async completions, performance test reporting).
```
Comment (stephentoub) [inline Documentation/debugging/windows-instructions.md]: I assume these are EventSources?  It'd be good to mention that.  There are also other EventSources in coreclr/corefx.  There are also other EventSources in coreclr/corefx; could you add those as well?  Here are the ones I'm aware of:
- https://github.com/dotnet/coreclr/blob/bc146608854d1db9cdbcc0b08029a87754e12b49/src/mscorlib/src/System/Diagnostics/Eventing/FrameworkEventSource.cs
- https://github.com/dotnet/coreclr/blob/bc146608854d1db9cdbcc0b08029a87754e12b49/src/mscorlib/src/System/Threading/Tasks/TPLETWProvider.cs
- https://github.com/dotnet/corefx/blob/ceae1d38c93e3f53210ad91884dabc1bd6d30a90/src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/DiagnosticSourceEventSource.cs
- https://github.com/dotnet/corefx/blob/c02d33b18398199f6acc17d375dab154e9a1df66/src/System.Collections.Concurrent/src/System/Collections/Concurrent/CDSCollectionETWBCLProvider.cs
- https://github.com/dotnet/corefx/blob/c02d33b18398199f6acc17d375dab154e9a1df66/src/System.Collections.Concurrent/src/System/Collections/Concurrent/CDSCollectionETWBCLProvider.cs
- https://github.com/dotnet/corefx/blob/c02d33b18398199f6acc17d375dab154e9a1df66/src/System.Threading/src/System/Threading/CDSsyncETWBCLProvider.cs
- https://github.com/dotnet/corefx/blob/c02d33b18398199f6acc17d375dab154e9a1df66/src/System.Linq.Parallel/src/System/Linq/Parallel/Utils/PLINQETWProvider.cs
- https://github.com/dotnet/corefx/blob/c02d33b18398199f6acc17d375dab154e9a1df66/src/System.Threading.Tasks.Parallel/src/System/Threading/Tasks/ParallelETWProvider.cs
- https://github.com/dotnet/corefx/blob/c02d33b18398199f6acc17d375dab154e9a1df66/src/System.Threading.Tasks.Dataflow/src/Internal/DataflowEtwProvider.cs

**Comment by stephentoub** [comment]
LGTM.  Thanks for putting this together.

File: Documentation/debugging/windows-instructions.md (hunk 2 above)
Comment (CIPop) [inline Documentation/debugging/windows-instructions.md]: Sure. Will do.

File: Documentation/debugging/windows-instructions.md (hunk 3 above)
Comment (CIPop) [inline Documentation/debugging/windows-instructions.md]: Thanks! I'll add these.

**Pull request merged**